### PR TITLE
Port int16 support to Mean kernel

### LINKED ISSUE #239: Port Int16 support to MEAN kernel
This issue tracks my work on porting Int16 support to the MEAN kernel in micro. @advaitjain 

It will be delivered as one PR with two commits (one for porting kernel, one for tests).

### PR DESCRIPTION
**Pull request by patriklaurell**
@advaitjain this PR ports int16 support to the Mean kernel.

BUG=Fixes #239

### CODE REVIEW

**Comment by patriklaurell** [comment]
@advaitjain ready for review

**Comment by advaitjain** [comment]
apologies for the delay. I will review and respond tomorrow.

File: tensorflow/lite/micro/kernels/reduce_test.cc [hunk 1]
```diff
@@ -262,6 +262,29 @@ TF_LITE_MICRO_TEST(MeanInt82DKeepDims) {
       output_zero_point, &params);
 }
 
+TF_LITE_MICRO_TEST(MeanInt162DKeepDims) {
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/reduce_test.cc:265]: Am I missing something or is it the case that the test cases are only tesing a single code branch
```cc
      } else if (op_data->input_zp == op_data->output_zp &&
                 op_data->input_scale == op_data->output_scale) {
```

Looks like this is not unique to the int16 port so I'm happy to make a Github issue to come back and fix this for the other types as well.

If you can add in additional test cases for int16 to the current PR, that would be appreciated.

**Review by advaitjain — CHANGES_REQUESTED** [review]

**Comment by patriklaurell** [comment]
@advaitjain The old tests cover the first two code paths
```cpp
if (params->keep_dims && special_case_4d_axes_1_and_2) {
```
and
```cpp
} else if (op_data->input_zp == op_data->output_zp &&
                 op_data->input_scale == op_data->output_scale) {
```
I added an additional test to cover the else clause.

**Review by advaitjain — APPROVED** [review]

**Pull request merged**